src/main/java/com/joejoe2/chat/models/PrivateChannel.java

```diff
@@ -73,6 +81,35 @@ public User anotherMember(User member) {
     return getMembers().stream().filter(u -> !member.getId().equals(u.getId())).findFirst().get();
   }
 
+  /**
+   * block or unblock the user
+   *
+   * @param user target user, cannot {@link #addMessage} until unblocked
+   * @param isBlock block or unblock
+   */
+  public void block(User user, boolean isBlock) {
```

Comment:
Is this project supposed to follow DDD?

Reply:
Originally not. But I would like to do so.

Reply:
I see. As a side note, implementing a DDD for small applications like chat services is not recommended. DDD can simplify much when it's a large enterprise SOA (or microservice) application with a strictly defined domain model bounded context.

From experience, this application is not a DDD candidate by any means.

Although taking into consideration that this is a pet project then "why not?" :) Let's give it a try. Then it will require a significant architectural adjustment. At least service layer removal.